go.mod

```diff
@@ -4,7 +4,7 @@ module github.com/cosmos/cosmos-sdk
 
 require (
 	cosmossdk.io/api v0.7.0
-	cosmossdk.io/collections v0.3.0
+	cosmossdk.io/collections v0.3.1-0.20230808102719-f04fefdc7a68
```

Comment:
@testinginprod, it would be great to get a new version before v0.50 final.
I remember you said you were waiting for something last time.

Reply:
waiting for go 1.21, this has improvements on type inference that should make certain things in collections better, this will cause some breaking changes on collections.

unfortunately, I'm not sure if this will happen before we cut v0.50 final. 

when there's intention to cut v0.50 final let me know, and I'll cut collections.